### PR TITLE
Display generated lot barcode

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR : 
- Adds the feature to show to the user the lot barcode as document

image

- Adds in the lot registry the barcode column

chrome_jCExftzSas

- Adds an option to search in the lot registry with the generated barcode

image

- Fix warning about expiration for assets

### CODE REVIEW

File: client/src/js/services/receipts/ReceiptService.js [hunk 1]
```diff
@@ -184,6 +184,14 @@ function ReceiptService($http, util, Language, AppCache, Session) {
   service.stockRequisitionReceipt = stockRequisitionReceipt;
   service.stockAdjustmentReport = stockAdjustmentReport;
   service.stockAggregateConsumptionReceipt = stockAggregateConsumptionReceipt;
+  service.lotBarcodeReceipt = lotBarcodeReceipt;
+
+  // lot barcode
+  function lotBarcodeReceipt(uuid, options) {
+    options.posReceipt = service.posReceipt;
+    const route = `/receipts/stock/lots/${uuid}/barcode`;
```
Comment (jniles) [inline client/src/js/services/receipts/ReceiptService.js:192]: Nice 👍

File: server/controllers/stock/core.js [hunk 2]
```diff
@@ -113,6 +113,12 @@ function getLotFilters(parameters) {
   filters.equals('trackingExpiration', 'tracking_expiration');
   filters.equals('stock_requisition_uuid', 'stock_requisition_uuid', 'm');
 
+  // barcode
+  filters.custom(
+    'barcode',
+    `CONCAT('LT', LEFT(HEX(l.uuid), 8)) = ?`,
```
Comment (jniles) [inline server/controllers/stock/core.js:119]: Nice 👍

File: server/controllers/stock/core.js [hunk 3]
```diff
@@ -892,7 +899,7 @@ function computeLotIndicators(inventories) {
         }
 
         lot.exhausted = lot.quantity <= 0;
-        lot.expired = !lot.exhausted && (lot.expiration_date < today);
+        lot.expired = !lot.exhausted && (lot.expiration_date < today && lot.tracking_expiration);
```
Comment (jniles) [inline server/controllers/stock/core.js:902]: Good catch!

File: server/models/migrations/next/migrate.sql [hunk 4]
```diff
@@ -9,3 +9,6 @@ UPDATE report SET `report_key` = 'analysis_auxiliary_cashbox', `title_key` = 'RE
 UPDATE unit SET
   `name` = 'Analysis of Cashbox', `key` = 'REPORT.ANALYSIS_AUX_CASHBOX.TITLE', `description` = 'Analysis of auxiliary cashbox', `path` = '/reports/analysis_auxiliary_cashbox'
   WHERE path = '/reports/analysis_auxiliary_cashboxes';
+
+-- regenerate barcodes for lots
```
Comment (jniles) [inline server/models/migrations/next/migrate.sql]: Do you need this?

**Review by jniles — COMMENTED** [review]
I thinik you don't need the `migrate.sql` lines.

Otherwise, looks good to me!

**Review by jmcameron — COMMENTED** [review]
I like this PR.  I scanned the code and it looks okay.  I tested it and it works nicely.

One small suggestion:  Since all the document has on it is the barcode, there is room to add more info.  For instance, I would suggest including the Inventory code.   Also, I suggest the using multiple lines for the comments.  For instance:

   Quinine Bichlorhydrate, sirop, 100mg base/5ml, 100ml, flacon, Unité - QUININE-B

would become:

     Quinine Bichlorhydrate, sirop, 100mg base/5ml, 100ml, flacon, Unité
     Inventory Code:  DORA_QUIN1S-_0,  Lot: QUININE-B
     [barcode]

Also (later) when we have fully support asset tracking in 'master', we may want to add some additional info if it is an asset.

File: server/controllers/stock/reports/stock/lot_barcode/lot_barcode.handlebars [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{{> head title="LOTS.BARCODE_FOR_LOT" }}
+
+<!-- body  -->
+<div class="text-center">
+  {{#if metadata.enterprise.settings.enable_barcodes}}
+    <small>{{ details.inventory_text }}</small><br>
```
Comment (jmcameron) [inline server/controllers/stock/reports/stock/lot_barcode/lot_barcode.handlebars]: I don't think the text needs to be "small" here.  I suggest removing the 'small' tags.  There is plenty of room on the document!

**Review by jmcameron — APPROVED** [review]
Thanks for the update.   I had one minor cosmetic suggestion (see the inline comments).  But I'm approving it.  If you want to remove the "small" html tags around the text, feel free to go ahead and do that before you merge it.

**Comment by mbayopanda** [comment]
@jmcameron changes landed

**Review by jmcameron — APPROVED** [review]
Bruce, this is ready to go.  I suggest that you rebase and then merge it yourself.

**Comment by jniles** [comment]
Let's let bors rebase!

bors r+

**Pull request merged**